### DISCUSSION TITLE
Custom JobDelegator Support In Quarkus

### DISCUSSION THREAD

**Discussion by arpitmotwani15** [quarkusio/quarkus#44018] [Q&A]
Hi Team ,

Do we have support for Using custom job delegate support for quarkus quartz extension like quartz (property name org.quartz.jobStore.driverDelegateClass) 

Looking at the code https://github.com/quarkusio/quarkus/blob/main/extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java#L136

It seems that the classnames  for Job Delegator are static 

Please can someone advice if it's possible to add a custom class for job delegator in quarkus quartz?

**Comment by arpitmotwani15** [comment]
Hi @mkouba please can you help with this 
Thanks ,
Arpit

**Reply by mkouba** [reply]
Hi Arpit, `org.quartz.jobStore.driverDelegateClass`, i.e. a custom delegate class, is not supported ATM. 

What does your custom delegate do? I mean, why is it needed?

**Reply by arpitmotwani15** [reply]
Hi Martin ,
Our custom delegate has some queries based on acquiring trigger on some custom conditions and also some custom business logic that we are doing in our application , are we supporting / planning to support something like this in quarkus-quartz ?

It is needed because in multiple replicas setup we would want certain jobs to be picked only on one replica

Right now it seems we are using quartz core queries to acquire trigger and they only use priority , next_fire_time to decide which trigger to pick next

Thanks,
Arpit

**Reply by mkouba** [reply]
> Our custom delegate has some queries based on acquiring trigger on some custom conditions and also some custom business logic that we are doing in our application , are we supporting / planning to support something like this in quarkus-quartz ?

That's quite some logic. I think that we could add a config property to specify a custom delegate with a big warning that it's the responsibility of the developer to use a delegate that extends one of the Quarkus wrappers (such as `QuarkusPostgreSQLDelegate`). WDYT @manovotn? 

> It is needed because in multiple replicas setup we would want certain jobs to be picked only on one replica

Shouldn't this be handled by Quartz automatically?

**Reply by manovotn** [reply]
> That's quite some logic. I think that we could add a config property to specify a custom delegate with a big warning that it's the responsibility of the developer to use a delegate that extends one of the Quarkus wrappers (such as QuarkusPostgreSQLDelegate). WDYT @manovotn?

Yes, we should be able to do that along with some validation so long as we can find the custom implementation class in the Jandex index.
I've created the following tracking issue - https://github.com/quarkusio/quarkus/issues/44050

**Comment by manovotn** [comment]
This feature was added as part of https://github.com/quarkusio/quarkus/pull/44064
@arpitmotwani15 you can try it right away in the main branch or wait for the next release.
